lib/mix/tasks/compile.make.ex

```diff
@@ -106,7 +111,8 @@ defmodule Mix.Tasks.Compile.ElixirMake do
   end
 
   defp build(config, task_args) do
-    exec      = Keyword.get(config, :make_executable, :default) |> os_specific_executable()
+    exec      = System.get_env("MAKE") ||
+    Keyword.get(config, :make_executable, :default) |> os_specific_executable()
```

Comment:
Let's keep this on one line and remove the pipe since it becomes cumbersome with the `||`:

```elixir
exec = System.get_env("MAKE") || os_specific_executable(Keyword.get(config, :make_executable, :default))
```
```